lambdas/check-coi/src/main/java/uk/gov/di/ipv/core/checkcoi/CheckCoiHandler.java

```diff
@@ -84,11 +99,41 @@ public Map<String, Object> handleRequest(ProcessRequest request, Context context
             var clientOAuthSession =
                     clientOAuthSessionDetailsService.getClientOAuthSession(
                             ipvSession.getClientOAuthSessionId());
-            LogHelper.attachGovukSigninJourneyIdToLogs(
-                    clientOAuthSession.getGovukSigninJourneyId());
+            var govukSigninJourneyId = clientOAuthSession.getGovukSigninJourneyId();
+            LogHelper.attachGovukSigninJourneyIdToLogs(govukSigninJourneyId);
 
             var userId = clientOAuthSession.getUserId();
 
+            var auditEventUser =
+                    new AuditEventUser(
+                            userId,
+                            ipvSession.getIpvSessionId(),
+                            govukSigninJourneyId,
+                            request.getIpAddress());
+
+            var coiSubjourneyType = ipvSession.getCoiSubjourneyType();
+            var coiCheckType =
+                    switch (coiSubjourneyType) {
+                        case GIVEN_NAMES_ONLY, GIVEN_NAMES_AND_ADDRESS -> CoiCheckType
+                                .LAST_NAME_AND_DOB;
+                        case FAMILY_NAME_ONLY, FAMILY_NAME_AND_ADDRESS -> CoiCheckType
+                                .GIVEN_NAMES_AND_DOB;
+                        case REVERIFICATION -> CoiCheckType.FULL_NAME_AND_DOB;
+                        case ADDRESS_ONLY -> {
```

Comment:
This is one questions. Since we're not expecting to do a Coi check for an address_only update details journey, is it ok to throw here instead of yield true? Especially now we're sending audit events

Reply:
Can we just remove `ADDRESS_ONLY` from the enum?

Reply:
It's still used for knowing which VCs to delete in the reset session identity handler

Reply:
Hmm, I guess we'll have to rework that as part of https://govukverify.atlassian.net/browse/PYIC-6582 - IMO we should be passing that in as a lambdaInput to the reset session handler as well.

(cc @Wynndow )

Reply:
Funnily enough I've been doing exactly that with the PR. Although given the audit event stuff we've just been discussing that might change.
https://github.com/govuk-one-login/ipv-core-back/pull/1977